uktrade/data-hub-frontend
# Add trade agreements to interactions

## Description of change

This PR adds trade agreement functionality to interactions, including a new component to implement a multiselect pattern found in the Nunjucks forms.

This relies on the following api PR: https://github.com/uktrade/data-hub-api/pull/3451

~~@ian-leggett mentioned that it would be good to improve the accessibility on this in order to help fix some accessibility issues elsewhere, I'll try to get to this when I'm free.~~ this will now be dealt with after this goes live as part of the broader piece of work on improving accessibility.  

## Test instructions

- Enable the "trade-agreement-interaction-v4-endpoint" feature flag in the API and switch to the branch in the aforementioned PR
- Navigate to a company, choose 'add interaction', select 'trade agreement'
- You should see trade agreement specific guidance text
- Select the 'trade agreements' service and another service
- You should see a question about trade agreements, answer yes
- Add some trade agreements

## Screenshots
### Before
![Screenie_6](https://user-images.githubusercontent.com/41294831/115705944-0c667e00-a365-11eb-8d94-4e0b587fe51a.png)
![Screenie_5](https://user-images.githubusercontent.com/41294831/115705947-0cff1480-a365-11eb-906f-32cfca932b14.png)

### After

![Screenie_1](https://user-images.githubusercontent.com/41294831/116247925-3eac1d00-a763-11eb-8f2d-ce84012b6b33.png)
![Screenie_7](https://user-images.githubusercontent.com/41294831/115706128-361fa500-a365-11eb-9a27-0b8e120e2d7f.png)


## Checklist

[//]: # "When submitting a PR make sure the code review guidelines have been satisfied.
https://github.com/uktrade/data-hub-frontend/blob/master/docs/Code%20review%20guidelines.md"
